chalk-solve/src/solve.rs

```diff
@@ -316,10 +225,32 @@ impl<I: Interner> Solver<I> {
                 expected_answers,
             } => {
                 let ops = SlgContextOps::new(program, *max_size, *expected_answers);
-                forest.solve(&ops, goal, || true)
+                ops.make_solution(goal, forest.iter_answers(&ops, goal), || true)
             }
             #[cfg(feature = "recursive-solver")]
-            SolverImpl::Recursive(ctx) => ctx.solver(program).solve_root_goal(goal).ok(),
+            SolverImpl::Recursive(ctx) => {
+                ctx.solver(program)
```

Comment:
*But* (continued from above still), this is a bit annoying.

Reply:
Yes.. that is indeed a bit annoying. Nevertheless if what you suggested happens (chalk engine + chalk_recursive depending on chalk solve), this one solver struct can go away correct?